jgrapht-core/src/main/java/org/jgrapht/alg/RankingPathElementList.java

```diff
@@ -343,11 +351,14 @@ private boolean isNotValidPath(RankingPathElement<V, E> prevPathElement, E edge)
         if (isGuardVertexDisconnected(prevPathElement)) {
             return true;
         }
-        if (externalPathValidator != null
-            && !externalPathValidator.isValidPath(prevPathElement, edge))
-        {
+        if (externalPathValidator != null && !externalPathValidator.isValidPath(
+            new GraphWalk<V, E>(
+                graph, getStartVertex(prevPathElement), prevPathElement.getVertex(),
```

Comment:
Last change in order to merge this one. Try to create the `GraphWalk` using only one traversal instead of two (one for the start vertex and one for the edge list).